src/runmode-pcap-folder.c

```diff
@@ -0,0 +1,265 @@
+/* Copyright (C) 2007-2010 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata-common.h"
+#include "tm-threads.h"
+#include "conf.h"
+#include "runmodes.h"
+#include "runmode-pcap-folder.h"
+#include "output.h"
+
+#include "detect-engine.h"
+#include "source-pcap-folder.h"
+
+#include "util-debug.h"
+#include "util-time.h"
+#include "util-cpu.h"
+#include "util-affinity.h"
+
+#include "util-runmodes.h"
+
+static const char *default_mode = NULL;
+
+const char *RunModeFolderPcapGetDefaultMode(void)
+{
+    return default_mode;
+}
+
+void RunModeFolderPcapRegister(void)
+{
+    RunModeRegisterNewRunMode(RUNMODE_PCAP_FOLDER, "single",
+                              "Single threaded pcap folder mode",
+                              RunModeFolderPcapSingle);
+    default_mode = "autofp";
+    RunModeRegisterNewRunMode(RUNMODE_PCAP_FOLDER, "autofp",
+                              "Multi threaded pcap folder mode.  Packets from "
+                              "each flow are assigned to a single detect thread, "
+                              "unlike \"pcap-folder-auto\" where packets from "
+                              "the same flow can be processed by any detect "
+                              "thread",
+                              RunModeFolderPcapAutoFp);
+
+    return;
+}
+
+/**
+ * \brief Single thread version of the Pcap folder processing.
+ */
+int RunModeFolderPcapSingle(void)
+{
+    const char *folder = NULL;
+    char tname[TM_THREAD_NAME_MAX];
+
+    if (ConfGet("pcap-folder.folder", &folder) == 0) {
+        SCLogError(SC_ERR_RUNMODE, "Failed retrieving pcap-folder from Conf");
+        exit(EXIT_FAILURE);
+    }
+
+    RunModeInitialize();
+    TimeModeSetOffline();
+
+    PcapFolderGlobalInit();
```

Comment:
With the exception of this line this file seems a complete copy of the regular pcap-file runmode. I'm not a fan of code duplication like this as it's likely to get out of sync at some point.